projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss

```diff
@@ -574,18 +592,122 @@
+        }
+
+        %cal-value,
+        %cal-value--label,
+        {
+            // 100 divided by the number of weekdays
+            min-width: 14.28%;
+
+            // Quantity Query selector
+            // 8 comes from the number of columns in the calendar
+            &:nth-last-child(8):first-child,
+            &:nth-last-child(8):first-child ~ * {
+                // 100 divided by the number of weekdays + weeks number column
+                min-width: 12.5% !important;
             }
         }
 
+        %cal-value-date--week-number {
+            padding-#{$right}: 0 !important;
+        }
+
+        %cal-value-date--week-number + %cal-value {
+            padding-left: 0 !important;
+        }
+
+        %label--week-number,
+        %cal-value-content--week-number {
+            position: relative;
+            color: --var($theme, 'week-number-color') !important;
+            border-radius: 0;
+            pointer-events: none;
+        }
+
+        %cal-value-date--week-number {
+            cursor: default;
+
+
+            &:hover,
+            &:focus {
+                %cal-value-content--week-number {
+                    color: --var($theme, 'week-number-color') !important;
+                }
+            }
+        }
+
+        %cal-value-content--week-number {
+            position: relative;
+            background: --var($theme, 'week-number-background') !important;
+            min-width: auto;
+
+            &:after {
+                display: none;
+            }
+
+            &:before {
+                content: '';
+                position: absolute;
+                width: 100%;
+                height: $cal-row-padding / 2;
+                left: 0;
+                top: 100%;
+                background: --var($theme, 'week-number-background') !important;
+            }
+        }
+
+        %label--week-number {
+            padding-right: 0 !important;
+            text-align: center;
+
+            span {
+                width: $cal-value-w;
+                position: relative;
+                background: --var($theme, 'week-number-background');
+
+                &:before,
+                &:after {
+                    content: '';
+                    position: absolute;
+                    width: 100%;
+                    background: --var($theme, 'week-number-background');
+                }
+
+                &:after {
+                    top: 100%;
+                    left: 0;
+                    @if $not-bootstrap-theme{
+                        height: ($cal-row-padding / 2) + rem(2px);
+                    } @else {
+                        height: $cal-row-padding + rem(5px);
+                    }
+                }
+
+                &:before {
+                    bottom: 100%;
+                    left: 0;
+                    height: $cal-row-padding;
+                    border-top-left-radius: $border-radius;
+                    border-top-right-radius: $border-radius;
```

Comment:
RTL.

Reply:
This is not affected by RTL. The radius is the same for all corners.